### ISSUE TITLE
[BUG] PathGraph, etc deprecations aren't exported

### ISSUE THREAD

**Issue by sbromberger** [JuliaGraphs/LightGraphs.jl#1232]
**Description of bug**
After the merge of #1223 `PathGraph` and other camel case constructors were deprecated in favor of their snake-case equivalents. Deprecations were listed in `src/SimpleGraphs/generators/deprecations.jl`, and the new snake case functions were exported in `src/LightGraphs/jl`,  but the old names were never exported top-level. This means that any code using `PathGraph` after this merge will fail as the name is no longer exported. This is a bug as it turns #1223 into a breaking change.

**How to reproduce**
```
using LightGraphs
g = PathGraph(5)
```

**Expected behavior**
I expected a deprecation warning.

**Actual behavior**
```
julia> g = PathGraph(5)
ERROR: UndefVarError: PathGraph not defined
Stacktrace:
 [1] top-level scope at REPL[2]:1
```

**Code demonstrating bug**
(see above)

**Version information**
```
julia> versioninfo()
Julia Version 1.2.0-DEV.661
Commit a0474d7cc2* (2019-04-07 11:23 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin18.5.0)
  CPU: Intel(R) Core(TM) i5-3210M CPU @ 2.50GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, ivybridge)
```
```
v1.2) pkg> status LightGraphs
    Status `~/.julia/environments/v1.2/Project.toml`
  [093fc24a] LightGraphs v1.2.0 [`~/.julia/dev/LightGraphs`]
  [2f01184e] SparseArrays 
```

**Additional context**
cc @matbesancon

**Comment by matbesancon** [comment]
Dang, the doc of `@deprecate` is misleading then:
```julia
help?> @deprecate
  @deprecate old new

  The first argument old is the signature of the deprecated method, the second
  one new is the call which replaces it. @deprecate exports the function.
```

**Comment by matbesancon** [comment]
@sbromberger closed by #1233
